apps/system/lockscreen/js/source/minute_clock_source.js

```diff
@@ -0,0 +1,55 @@
+/* global SourceEvent */
+'use strict';
+
+/**
+ * A source fire events every clock minutes.
+ **/
+(function(exports) {
+  var MinuteClockSource = function(configs) {
+    this.configs = {
+      type: configs.type,
+      interval: 60000       // one minute.
+    };
+    this._id = null;
+    this._forwardTo = null;
+    // Some API you just can't bind it with the object,
+    // but a function.
+    this.onchange = this.onchange.bind(this);
+  };
+
+  MinuteClockSource.prototype.start = function(forwardTo) {
+    this._forwardTo = forwardTo;
+    var seconds = (new Date()).getSeconds();
+    // If it's at the second 0th of the minute, immediate start to tick.
+    var leftMilliseconds = (0 === seconds) ? 0 : (60 - seconds) * 1000;
```

Comment:
I don't think this is correct -- The user of `MinuteClockSource` is responsible of getting the time right at the function loop of creating the source.

At least you want to account an unlikely situation where the second when we do `Date.now()` else where is not the same here should the code runs very slowly....